source/common/network/io_socket_handle_impl.h

```diff
@@ -64,8 +64,20 @@ class IoSocketHandleImpl : public IoHandle, protected Logger::Loggable<Logger::I
   absl::optional<int> domain() override;
   Address::InstanceConstSharedPtr localAddress() override;
   Address::InstanceConstSharedPtr peerAddress() override;
-  Event::FileEventPtr createFileEvent(Event::Dispatcher& dispatcher, Event::FileReadyCb cb,
-                                      Event::FileTriggerType trigger, uint32_t events) override;
+  void initializeFileEvent(Event::Dispatcher& dispatcher, Event::FileReadyCb cb,
+                           Event::FileTriggerType trigger, uint32_t events) override;
+
+  IoHandlePtr duplicate() override;
+
+  void activateFileEvents(uint32_t events) override;
+  void enableFileEvents(uint32_t events) override;
+
+  void resetFileEvents() override {
+    if (file_event_) {
+      file_event_.reset();
+    }
```

Comment:
nit: you could just call reset, it is a noop if the file_event _ is already null

Reply:
A bit of irony that the only null check was for the one function that did not actually need it. Fixed in 13ff61610c48391cf70ae52dbc51b2d88c2132dd